src/game_api/entities_items.hpp

```diff
@@ -65,14 +75,24 @@ class VladsCape : public Cape
     bool can_double_jump;
 };
 
-class Mattock : public Movable
+class Purchasable : public Movable
+{
+    virtual void buy() = 0;
+};
+
+class DummyPurchasableEntity : public Purchasable
+{
+    virtual Entity* switch_entities(void*) = 0; // switches the purchasable cape with normal one
+};
+
+class Mattock : public Purchasable
 {
   public:
     uint8_t remaining;
 };
 
-class Gun : public Movable
-{
+class Gun : public Purchasable
+{ // BIG NOTE: scepter is not a Purchasable, but that's the only Gun that isn't so i din't bother to make new type for it
```

Comment:
Wouldn't this expose a non existent `buy` virtual for the scepter then? I imagine anyone calling that would be undefined and bad?

Reply:
only stuff in src/game_api/script is exposed to lua, and also we decide what to expose, it's not automatic

until i know this function can be used in some way or another, there is no reason to add another pointless type for one entity, i added the rest only cuz it was relatively easy